src/swtpm/common.c

```diff
@@ -620,16 +633,30 @@ parse_tpmstate_options(char *options, char **tpmstatedir, mode_t *mode)
     }
 
     directory = option_get_string(ovs, "dir", NULL);
-    if (!directory) {
-        logprintf(STDERR_FILENO,
-                  "The file parameter is required for the tpmstate option.\n");
-        goto error;
-    }
     *mode = option_get_mode_t(ovs, "mode", 0640);
+    backend = option_get_string(ovs, "backend", NULL);
+    backend_uri = option_get_string(ovs, "backend_uri", NULL);
 
-    *tpmstatedir = strdup(directory);
-    if (!*tpmstatedir) {
-        logprintf(STDERR_FILENO, "Out of memory.");
+    if (directory) {
+        *tpmstatedir = strdup(directory);
+        if (!*tpmstatedir) {
+            logprintf(STDERR_FILENO, "Out of memory.");
+            goto error;
+        }
+    } else if (backend && backend_uri) {
+        *tpmstatebackend = strdup(backend);
+        if (!*tpmstatebackend) {
+            logprintf(STDERR_FILENO, "Out of memory.");
+            goto error;
+        }
+        *tpmstatebackend_uri = strdup(backend_uri);
+        if (!*tpmstatebackend_uri) {
+            logprintf(STDERR_FILENO, "Out of memory.");
+            goto error;
+        }
```

Comment:
error label should free *tpmstatebackend.
You could also move the two strdup's together and then do
`if (!*tpmstatebackend || !*tpmstatebackenduri) { ...`
to make this a bit shorter,

Reply:
Thanks, let me use your shorter way.
`*tpmstatebackend` is also freed in caller (`handle_tpmstate_options()`) but it may be clearer to add free in error label too. I'll add them.